### ISSUE TITLE
Installing firedrake -- problems with petsc

### ISSUE THREAD

**Issue by stavener** [firedrakeproject/firedrake#1368]
Cloning petsc

Failed to clone petsc using ssh, falling back to https.
Failed to clone petsc branch firedrake.
Traceback (most recent call last):
  File "firedrake-install", line 525, in git_clone
    check_call(["git", "clone", "-q", git_url(plain_url, "ssh")])
  File "firedrake-install", line 438, in check_call
    log.debug(subprocess.check_output(arguments, stderr=subprocess.STDOUT, env=env).decode())
  File "/usr/lib/python3.6/subprocess.py", line 336, in check_output
    **kwargs).stdout
  File "/usr/lib/python3.6/subprocess.py", line 418, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['git', 'clone', '-q', 'git@github.com:firedrakeproject/petsc.git']' returned non-zero exit status 128.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "firedrake-install", line 1117, in <module>
    packages = clone_dependencies("PyOP2") + packages
  File "firedrake-install", line 564, in clone_dependencies
    deps.append(git_clone(dep.strip()))
  File "firedrake-install", line 531, in git_clone
    check_call(["git", "clone", "-q", git_url(plain_url, "https")])
  File "firedrake-install", line 438, in check_call
    log.debug(subprocess.check_output(arguments, stderr=subprocess.STDOUT, env=env).decode())
  File "/usr/lib/python3.6/subprocess.py", line 336, in check_output
    **kwargs).stdout
  File "/usr/lib/python3.6/subprocess.py", line 418, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['git', 'clone', '-q', 'https://github.com/firedrakeproject/petsc.git']' returned non-zero exit status 128.

Install log saved in /home/tavener/firedrake-install.log
tavener@surface-tavener:~$

**Comment by stavener** [comment]
Installing firedrake on a windows 10 machine. The install log is here.
[firedrake-install.log](https://github.com/firedrakeproject/firedrake/files/2814648/firedrake-install.log)
All going swimmingly following the until the installation directions until the loading of petsc. Can you help please?

**Comment by wence-** [comment]
Looks like your internet connection had a hiccough?
```
2019-01-30 21:56:42,116 DEBUG  Running command 'git clone -q https://github.com/firedrakeproject/petsc.git'
2019-01-30 21:57:13,811 DEBUG  error: RPC failed; curl 56 GnuTLS recv error (-9): A TLS packet with unexpected length was received.
fatal: The remote end hung up unexpectedly
fatal: early EOF
fatal: index-pack failed

2019-01-30 21:57:13,812 ERROR  Failed to clone petsc branch firedrake.
```

**Comment by wence-** [comment]
Can you just try again? You'll have to delete the "firedrake" directory that was created since we do not have robust mechanisms for resuming from partial installations that somehow failed partway through.

Thanks.

**Comment by stavener** [comment]
Will try. I discovered that earlier. Will use rm -rf firedrake again. Thank you.

**Comment by stavener** [comment]
BTW, I am unsure why it is failing to clone using ssh, but I expect it is a pwd issue.

**Comment by stavener** [comment]
Same issue as before, but I have an idea. When I ran the test cases, I got a message that petsc4py was not present, so I installed it using pip install. I was curious, because I had not had to do this on another windows 10 machine for which my installation was successful. (A little knowledge is a dangerous thing!) I have now purged petsc4py, removed all the firedrake files and started again.

**Comment by stavener** [comment]
Thank you. The installation procedure went smoothly this morning with a better wifi connection.  

However,

> make alltest

resulted in the following...

```
(firedrake) tavener@surface-tavener:~/firedrake/src/firedrake$ make alltest
    Building extension modules
    Linting firedrake codebase
firedrake/utility_meshes.py:797:13: E117 over-indented
firedrake/utility_meshes.py:1153:13: E117 over-indented
firedrake/variational_solver.py:273:12: F632 use ==/!= to compare str, bytes, and int literals
Makefile:8: recipe for target 'lint' failed
make: *** [lint] Error 1
```
Similarly, 

> make shorttest

resulted in a similar error.  Conversely, 

> make test_demos

was successful and 

> make test_regression 

is running smoothly thus far (25% complete). 

In case you are wondering, as I am...

```
tavener@surface-tavener:~$ find . -name flake8 -print
./firedrake/bin/flake8
./firedrake/lib/python3.6/site-packages/flake8
```

**Comment by wence-** [comment]
Ah, the linting (that is, checking that our code obeys our own coding standards) failed in the first step. This looks to be because the tool we use has updated its ruleset and is now stricter than it was last week.

```
(firedrake) tavener@surface-tavener:~/firedrake/src/firedrake$ make alltest
    Building extension modules
    Linting firedrake codebase
firedrake/utility_meshes.py:797:13: E117 over-indented
firedrake/utility_meshes.py:1153:13: E117 over-indented
firedrake/variational_solver.py:273:12: F632 use ==/!= to compare str, bytes, and int literals
Makefile:8: recipe for target 'lint' failed
make: *** [lint] Error 1
```

On the other hand, this is only a stylistic thing, and does not affect the running of Firedrake, so if the test suite passes otherwise (as it appears to be doing so far), then all is well. (But we should fix the code style errors).

I'll close this issue,  but please do reopen one if you get any other errors.

Thanks!

**Comment by dham** [comment]
Note. that these lint errors are fixed in #1367
